src/System.IO.Pipelines/System/IO/Pipelines/ReadCursor.cs

```diff
@@ -285,12 +335,27 @@ public override int GetHashCode()
             return ((int)shift5 + h1) ^ h2;
         }
 
-        internal bool GreaterOrEqual(ReadCursor other)
+        internal static bool GreaterOrEqual(BufferSegment start, int startIndex, BufferSegment end, int endIndex)
         {
             // other.Segment.RunningLength + other.Index  - other.Segment.Start <= Segment.RunningLength + Index- Segment.Start
             // fliped to avoid overflows
 
-            return other.Segment.RunningLength - Index - other.Segment.Start <= Segment.RunningLength - other.Index - Segment.Start;
+            return end.RunningLength - startIndex - end.Start <= start.RunningLength - endIndex - start.Start;
+        }
+
+        internal BufferSegment GetSegment()
+        {
+            if (Segment == null)
+            {
+                return null;
+            }
+            if (Segment is BufferSegment bufferSegment)
+            {
+                return bufferSegment;
+            }
+
+            ThrowWrongType();
+            return default;
```

Comment:
Null not good enough? 😬

Reply:
I use `default` when I want to convey "we don't care what's returned". I like the pattern and how it works for both reference and value types.